app/graph/types/team_type.rb

```diff
@@ -400,4 +400,56 @@ def statistics(period:, language: nil, platform: nil)
     return nil unless User.current&.is_admin
     TeamStatistics.new(object, period, language, platform)
   end
+
+  field :bot_query, [TiplineSearchResultType], null: true do
+    argument :search_text, GraphQL::Types::String, required: true
+  end
+
+  def bot_query(search_text:)
+    unless User.current&.is_admin
+      raise GraphQL::ExecutionError, "You do not have permission to perform this action"
+    end
+
+    return nil unless search_text
```

Comment:
Not a blocker: Not needed, since `search_text` is a required argument. This method won't even be executed if `search_text` is not provided.